Forwarder heal tests are unstable

## Failure Logs
https://github.com/networkservicemesh/sdk/runs/2378624756?check_suite_focus=true#step:5:125
```
--- FAIL: TestNSMGR_HealLocalForwarder (10.03s)
    server_heal_test.go:230: 
        	Error Trace:	server_heal_test.go:230
        	            				server_heal_test.go:142
        	Error:      	Received unexpected error:
        	            	rpc error: code = DeadlineExceeded desc = context deadline exceeded
        	            	Error returned from api/pkg/api/networkservice/networkServiceClient.Request
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*beginTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:62
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:71
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*endTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:99
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:71
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/common/authorize.(*authorizeClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/common/authorize/client.go:53
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*beginTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:58
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*endTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:99
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:71
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/utils/metadata.(*metaDataClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/utils/metadata/client.go:44
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*beginTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:58
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*endTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:99
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:71
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/common/refresh.(*refreshClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/common/refresh/client.go:56
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*beginTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:58
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*endTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:99
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:71
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/common/heal.(*healClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/common/heal/client.go:48
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*beginTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:58
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*endTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:99
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:71
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/common/serialize.(*serializeClient).Request.func1
        	            		D:/a/sdk/sdk/pkg/networkservice/common/serialize/client.go:45
        	            	github.com/networkservicemesh/sdk/pkg/tools/multiexecutor.(*MultiExecutor).AsyncExec.func2.1
        	            		D:/a/sdk/sdk/pkg/tools/multiexecutor/multi_executor.go:54
        	            	github.com/edwarnicke/serialize.(*Executor).process
        	            		C:/Users/runneradmin/go/pkg/mod/github.com/edwarnicke/serialize@v1.0.7/serialize.go:68
        	Test:       	TestNSMGR_HealLocalForwarder
```

https://github.com/networkservicemesh/sdk/runs/2378343352?check_suite_focus=true#step:5:140
```
--- FAIL: TestNSMGR_HealRemoteForwarder (10.04s)
    server_heal_test.go:228: 
        	Error Trace:	server_heal_test.go:228
        	            				server_heal_test.go:173
        	Error:      	Received unexpected error:
        	            	rpc error: code = DeadlineExceeded desc = context deadline exceeded
        	            	Error returned from api/pkg/api/networkservice/networkServiceClient.Request
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*beginTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:62
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:71
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*endTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:99
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:71
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/common/authorize.(*authorizeClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/common/authorize/client.go:53
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*beginTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:58
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*endTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:99
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:71
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/utils/metadata.(*metaDataClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/utils/metadata/client.go:44
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*beginTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:58
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*endTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:99
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:71
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/common/refresh.(*refreshClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/common/refresh/client.go:56
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*beginTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:58
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*endTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:99
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:71
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/common/heal.(*healClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/common/heal/client.go:48
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*beginTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:58
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:69
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/trace.(*endTraceClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/trace/client.go:99
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/core/next.(*nextClient).Request
        	            		D:/a/sdk/sdk/pkg/networkservice/core/next/client.go:71
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/common/serialize.(*serializeClient).Request.func1
        	            		D:/a/sdk/sdk/pkg/networkservice/common/serialize/client.go:45
        	            	github.com/networkservicemesh/sdk/pkg/tools/multiexecutor.(*MultiExecutor).AsyncExec.func2.1
        	            		D:/a/sdk/sdk/pkg/tools/multiexecutor/multi_executor.go:54
        	            	github.com/edwarnicke/serialize.(*Executor).process
        	            		C:/Users/runneradmin/go/pkg/mod/github.com/edwarnicke/serialize@v1.0.7/serialize.go:68
        	Test:       	TestNSMGR_HealRemoteForwarder
```

--- a/pkg/networkservice/chains/nsmgr/heal_test.go
+++ b/pkg/networkservice/chains/nsmgr/heal_test.go
@@ -43,8 +43,6 @@ func TestNSMGR_HealEndpoint(t *testing.T) {
 }
 
 func TestNSMGR_HealEndpointRestored(t *testing.T) {
-	t.Skip("https://github.com/networkservicemesh/sdk/issues/840")
-
 	testNSMGRHealEndpoint(t, true)
 }
 
@@ -63,13 +61,12 @@ func testNSMGRHealEndpoint(t *testing.T, restored bool) {
 	require.NoError(t, err)
 
 	nseReg := defaultRegistryEndpoint(nsReg.Name)
-	nseReg.ExpirationTime = timestamppb.New(time.Now().Add(time.Second))
+	nseReg.ExpirationTime = timestamppb.New(time.Now().Add(tokenTimeout(restored)))
 
-	nseCtx, nseCtxCancel := context.WithTimeout(context.Background(), time.Second)
-	defer nseCtxCancel()
+	nseCtx, nseCtxCancel := context.WithCancel(context.Background())
 
 	counter := &counterServer{}
-	nse, err := domain.Nodes[0].NewEndpoint(nseCtx, nseReg, sandbox.GenerateExpiringToken(time.Second), counter)
+	nse, err := domain.Nodes[0].NewEndpoint(nseCtx, nseReg, sandbox.GenerateExpiringToken(tokenTimeout(restored)), counter)
 	require.NoError(t, err)
 
 	request := defaultRequest(nsReg.Name)
@@ -113,16 +110,14 @@ func testNSMGRHealEndpoint(t *testing.T, restored bool) {
 }
 
 func TestNSMGR_HealLocalForwarder(t *testing.T) {
-	t.Skip("https://github.com/networkservicemesh/sdk/issues/845")
-
-	forwarderCtx, forwarderCtxCancel := context.WithTimeout(context.Background(), time.Second)
+	forwarderCtx, forwarderCtxCancel := context.WithCancel(context.Background())
 	defer forwarderCtxCancel()
 
 	customConfig := []*sandbox.NodeConfig{
 		nil,
 		{
 			ForwarderCtx:               forwarderCtx,
-			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(time.Second),
+			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout(false)),
 		},
 	}
 
@@ -130,16 +125,14 @@ func TestNSMGR_HealLocalForwarder(t *testing.T) {
 }
 
 func TestNSMGR_HealLocalForwarderRestored(t *testing.T) {
-	t.Skip("https://github.com/networkservicemesh/sdk/issues/840")
-
-	forwarderCtx, forwarderCtxCancel := context.WithTimeout(context.Background(), time.Second)
+	forwarderCtx, forwarderCtxCancel := context.WithCancel(context.Background())
 	defer forwarderCtxCancel()
 
 	customConfig := []*sandbox.NodeConfig{
 		nil,
 		{
 			ForwarderCtx:               forwarderCtx,
-			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(time.Second),
+			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout(true)),
 		},
 	}
 
@@ -147,15 +140,13 @@ func TestNSMGR_HealLocalForwarderRestored(t *testing.T) {
 }
 
 func TestNSMGR_HealRemoteForwarder(t *testing.T) {
-	t.Skip("https://github.com/networkservicemesh/sdk/issues/845")
-
-	forwarderCtx, forwarderCtxCancel := context.WithTimeout(context.Background(), time.Second)
+	forwarderCtx, forwarderCtxCancel := context.WithCancel(context.Background())
 	defer forwarderCtxCancel()
 
 	customConfig := []*sandbox.NodeConfig{
 		{
 			ForwarderCtx:               forwarderCtx,
-			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(time.Second),
+			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout(false)),
 		},
 	}
 
@@ -163,15 +154,13 @@ func TestNSMGR_HealRemoteForwarder(t *testing.T) {
 }
 
 func TestNSMGR_HealRemoteForwarderRestored(t *testing.T) {
-	t.Skip("https://github.com/networkservicemesh/sdk/issues/840")
-
-	forwarderCtx, forwarderCtxCancel := context.WithTimeout(context.Background(), time.Second)
+	forwarderCtx, forwarderCtxCancel := context.WithCancel(context.Background())
 	defer forwarderCtxCancel()
 
 	customConfig := []*sandbox.NodeConfig{
 		{
 			ForwarderCtx:               forwarderCtx,
-			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(time.Second),
+			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout(true)),
 		},
 	}
 
@@ -258,17 +247,14 @@ func testNSMGRHealForwarder(t *testing.T, nodeNum int, restored bool, customConf
 }
 
 func TestNSMGR_HealRemoteNSMgrRestored(t *testing.T) {
-	t.Skip("https://github.com/networkservicemesh/sdk/issues/840")
-
-	nsmgrCtx, nsmgrCtxCancel := context.WithTimeout(context.Background(), time.Second)
+	nsmgrCtx, nsmgrCtxCancel := context.WithCancel(context.Background())
 	defer nsmgrCtxCancel()
 
 	customConfig := []*sandbox.NodeConfig{
 		{
-			NsmgrCtx:                   nsmgrCtx,
-			NsmgrGenerateTokenFunc:     sandbox.GenerateExpiringToken(time.Second),
-			ForwarderCtx:               nsmgrCtx,
-			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(time.Second),
+			NsmgrCtx: nsmgrCtx,
+			// FIXME: it should be a restored case, but it doesn't work as a restored case
+			NsmgrGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout(false)),
 		},
 	}
 
@@ -318,9 +304,10 @@ func testNSMGRHealNSMgr(t *testing.T, nodeNum int, customConfig []*sandbox.NodeC
 	domain.AddResources(restoredNSMgrResources)
 
 	forwarderReg := &registry.NetworkServiceEndpoint{
-		Name: "forwarder-restored",
+		Name: "forwarder",
+		Url:  domain.Nodes[nodeNum].Forwarder[0].URL.String(),
 	}
-	_, err = domain.Nodes[nodeNum].NewForwarder(ctx, forwarderReg, sandbox.GenerateTestToken)
+	_, err = domain.Nodes[nodeNum].ForwarderRegistryClient.Register(ctx, forwarderReg)
 	require.NoError(t, err)
 
 	nseReg.Url = nse.URL.String()
@@ -349,14 +336,12 @@ func testNSMGRHealNSMgr(t *testing.T, nodeNum int, customConfig []*sandbox.NodeC
 func TestNSMGR_HealRemoteNSMgr(t *testing.T) {
 	t.Cleanup(func() { goleak.VerifyNone(t) })
 
-	nsmgrCtx, nsmgrCtxCancel := context.WithTimeout(context.Background(), time.Second)
-	defer nsmgrCtxCancel()
+	nsmgrCtx, nsmgrCtxCancel := context.WithCancel(context.Background())
+
 	customConfig := []*sandbox.NodeConfig{
 		{
-			NsmgrCtx:                   nsmgrCtx,
-			NsmgrGenerateTokenFunc:     sandbox.GenerateExpiringToken(time.Second),
-			ForwarderCtx:               nsmgrCtx,
-			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(time.Second),
+			NsmgrCtx:               nsmgrCtx,
+			NsmgrGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout(false)),
 		},
 	}
 
@@ -465,9 +450,16 @@ func TestNSMGR_CloseHeal(t *testing.T) {
 	require.NoError(t, ctx.Err())
 }
 
+func tokenTimeout(restored bool) time.Duration {
+	if restored {
+		return time.Hour
+	}
+	return 3 * time.Second
+}
+
 func checkSecondRequestsReceived(requestsDone func() int) func() bool {
 	return func() bool {
-		return requestsDone() == 2
+		return requestsDone() >= 2
 	}
 }
 

--- a/pkg/networkservice/chains/nsmgr/heal_test.go
+++ b/pkg/networkservice/chains/nsmgr/heal_test.go
@@ -252,9 +252,8 @@ func TestNSMGR_HealRemoteNSMgrRestored(t *testing.T) {
 
 	customConfig := []*sandbox.NodeConfig{
 		{
-			NsmgrCtx: nsmgrCtx,
-			// FIXME: it should be a restored case, but it doesn't work as a restored case
-			NsmgrGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout(false)),
+			NsmgrCtx:               nsmgrCtx,
+			NsmgrGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout(true)),
 		},
 	}
 
@@ -262,7 +261,11 @@ func TestNSMGR_HealRemoteNSMgrRestored(t *testing.T) {
 }
 
 func testNSMGRHealNSMgr(t *testing.T, nodeNum int, customConfig []*sandbox.NodeConfig, nsmgrCtxCancel context.CancelFunc) {
+	// Restore NSMgr test cases cannot work without registry healing.
+	t.Skip("https://github.com/networkservicemesh/sdk/issues/713")
+
 	t.Cleanup(func() { goleak.VerifyNone(t) })
+
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
 
@@ -277,10 +280,8 @@ func testNSMGRHealNSMgr(t *testing.T, nodeNum int, customConfig []*sandbox.NodeC
 	nsReg, err := domain.Nodes[0].NSRegistryClient.Register(ctx, defaultRegistryService())
 	require.NoError(t, err)
 
-	nseReg := defaultRegistryEndpoint(nsReg.Name)
-
 	counter := &counterServer{}
-	nse, err := domain.Nodes[0].NewEndpoint(ctx, nseReg, sandbox.GenerateTestToken, counter)
+	_, err = domain.Nodes[0].NewEndpoint(ctx, defaultRegistryEndpoint(nsReg.Name), sandbox.GenerateTestToken, counter)
 	require.NoError(t, err)
 
 	request := defaultRequest(nsReg.Name)
@@ -303,18 +304,6 @@ func testNSMGRHealNSMgr(t *testing.T, nodeNum int, customConfig []*sandbox.NodeC
 	domain.Nodes[nodeNum].NSMgr = restoredNSMgrEntry
 	domain.AddResources(restoredNSMgrResources)
 
-	forwarderReg := &registry.NetworkServiceEndpoint{
-		Name: "forwarder",
-		Url:  domain.Nodes[nodeNum].Forwarder[0].URL.String(),
-	}
-	_, err = domain.Nodes[nodeNum].ForwarderRegistryClient.Register(ctx, forwarderReg)
-	require.NoError(t, err)
-
-	nseReg.Url = nse.URL.String()
-	_, err = domain.Nodes[nodeNum].EndpointRegistryClient.Register(ctx, nseReg)
-	require.NoError(t, err)
-
-	// Wait Cross NSE expired and reconnecting through the new Cross NSE
 	require.Eventually(t, checkSecondRequestsReceived(func() int {
 		return int(atomic.LoadInt32(&counter.Requests))
 	}), timeout, tick)
@@ -380,7 +369,7 @@ func TestNSMGR_HealRemoteNSMgr(t *testing.T) {
 	_, err = domain.Nodes[2].NewEndpoint(ctx, nseReg2, sandbox.GenerateTestToken, counter)
 	require.NoError(t, err)
 
-	// Wait Cross NSE expired and reconnecting through the new Cross NSE
+	// Wait NSMgr expired and reconnecting through the new NSMgr
 	require.Eventually(t, checkSecondRequestsReceived(counter.UniqueRequests), timeout, tick)
 	require.Equal(t, 2, counter.UniqueRequests())
 

--- a/pkg/networkservice/chains/nsmgr/heal_test.go
+++ b/pkg/networkservice/chains/nsmgr/heal_test.go
@@ -34,8 +34,9 @@ import (
 )
 
 const (
-	tick    = 10 * time.Millisecond
-	timeout = 10 * time.Second
+	tick         = 10 * time.Millisecond
+	timeout      = 10 * time.Second
+	tokenTimeout = 5 * time.Second
 )
 
 func TestNSMGR_HealEndpoint(t *testing.T) {
@@ -61,12 +62,19 @@ func testNSMGRHealEndpoint(t *testing.T, restored bool) {
 	require.NoError(t, err)
 
 	nseReg := defaultRegistryEndpoint(nsReg.Name)
-	nseReg.ExpirationTime = timestamppb.New(time.Now().Add(tokenTimeout(restored)))
+	if !restored {
+		nseReg.ExpirationTime = timestamppb.New(time.Now().Add(tokenTimeout))
+	}
 
 	nseCtx, nseCtxCancel := context.WithCancel(context.Background())
 
 	counter := &counterServer{}
-	nse, err := domain.Nodes[0].NewEndpoint(nseCtx, nseReg, sandbox.GenerateExpiringToken(tokenTimeout(restored)), counter)
+	var nse *sandbox.EndpointEntry
+	if restored {
+		nse, err = domain.Nodes[0].NewEndpoint(nseCtx, nseReg, sandbox.GenerateTestToken, counter)
+	} else {
+		nse, err = domain.Nodes[0].NewEndpoint(nseCtx, nseReg, sandbox.GenerateExpiringToken(tokenTimeout), counter)
+	}
 	require.NoError(t, err)
 
 	request := defaultRequest(nsReg.Name)
@@ -117,7 +125,7 @@ func TestNSMGR_HealLocalForwarder(t *testing.T) {
 		nil,
 		{
 			ForwarderCtx:               forwarderCtx,
-			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout(false)),
+			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout),
 		},
 	}
 
@@ -132,7 +140,7 @@ func TestNSMGR_HealLocalForwarderRestored(t *testing.T) {
 		nil,
 		{
 			ForwarderCtx:               forwarderCtx,
-			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout(true)),
+			ForwarderGenerateTokenFunc: sandbox.GenerateTestToken,
 		},
 	}
 
@@ -146,7 +154,7 @@ func TestNSMGR_HealRemoteForwarder(t *testing.T) {
 	customConfig := []*sandbox.NodeConfig{
 		{
 			ForwarderCtx:               forwarderCtx,
-			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout(false)),
+			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout),
 		},
 	}
 
@@ -160,7 +168,7 @@ func TestNSMGR_HealRemoteForwarderRestored(t *testing.T) {
 	customConfig := []*sandbox.NodeConfig{
 		{
 			ForwarderCtx:               forwarderCtx,
-			ForwarderGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout(true)),
+			ForwarderGenerateTokenFunc: sandbox.GenerateTestToken,
 		},
 	}
 
@@ -253,7 +261,7 @@ func TestNSMGR_HealRemoteNSMgrRestored(t *testing.T) {
 	customConfig := []*sandbox.NodeConfig{
 		{
 			NsmgrCtx:               nsmgrCtx,
-			NsmgrGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout(true)),
+			NsmgrGenerateTokenFunc: sandbox.GenerateTestToken,
 		},
 	}
 
@@ -330,7 +338,7 @@ func TestNSMGR_HealRemoteNSMgr(t *testing.T) {
 	customConfig := []*sandbox.NodeConfig{
 		{
 			NsmgrCtx:               nsmgrCtx,
-			NsmgrGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout(false)),
+			NsmgrGenerateTokenFunc: sandbox.GenerateExpiringToken(tokenTimeout),
 		},
 	}
 
@@ -437,13 +445,6 @@ func TestNSMGR_CloseHeal(t *testing.T) {
 	}, timeout, tick)
 
 	require.NoError(t, ctx.Err())
-}
-
-func tokenTimeout(restored bool) time.Duration {
-	if restored {
-		return time.Hour
-	}
-	return 3 * time.Second
 }
 
 func checkSecondRequestsReceived(requestsDone func() int) func() bool {

